sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -989,14 +994,25 @@ private class BulkWriter(container: CosmosAsyncContainer,
                   numberOfIntervalsWithIdenticalActiveOperationSnapshots
                 )
 
-                if (numberOfIntervalsWithIdenticalActiveOperationSnapshots.get == 1L &&
-                    // pending retries only tracks the time to enqueue
-                    // the retry - so this should never take longer than 1 minute
-                    pendingRetriesSnapshot == 0L &&
-                    Scannable.from(bulkInputEmitter).scan(Scannable.Attr.BUFFERED) >0) {
+                if (numberOfIntervalsWithIdenticalActiveOperationSnapshots.get > 0L) {
```

Comment:
just to confirm, so we also want to loose how many times we gonna requeue here?

Reply:
yes, I doubt it helps at that point - but won't hurt either.